modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -1075,6 +1083,34 @@ public IgniteConfiguration setSqlQueryHistorySize(int size) {
         return this;
     }
 
+    /**
+     * Defines the default query timeout.
+     *
+     * Defaults to {@link #DFLT_QRY_TIMEOUT}.
+     * {@code 0} means there is no timeout (this
+     * is a default value)
+     *
+     * @return Default query timeout.
+     */
+    public long getDefaultQueryTimeout() {
+        return dfltQryTimeout;
+    }
+
+    /**
+     * Sets timeout in milliseconds for default query timeout.
+     * {@code 0} means there is no timeout (this
+     * is a default value)
+     *
+     * @param dfltQryTimeout Timeout in milliseconds.
+     * @return {@code this} for chaining.
+     */
+    public IgniteConfiguration setDefaultQueryTimeout(long dfltQryTimeout) {
+        A.ensure(dfltQryTimeout >= 0 && dfltQryTimeout <= Integer.MAX_VALUE, "default query timeout value should be valid Integer.");
```

Comment:
please remove dfltQryTimeout <= Integer.MAX_VALUE, it's useless check, because it's always "true".

Reply:
@mcherkasov, not quite. `dfltQryTimeout` is long and can be greater than `Integer.MAX_VALUE`